applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -1069,22 +1108,7 @@ static void connect(const void *subscriber_id, uint8_t report_id)
 	rs->subscriber = subscriber;
 	rs->state = STATE_CONNECTED_IDLE;
 	rs->report_id = report_id;
-
-	/* Connect with apriopriete report data. */
-	uint8_t report_data_id = report_id;
-	switch (report_id) {
-	case REPORT_ID_BOOT_MOUSE:
-		report_data_id = REPORT_ID_MOUSE;
-		break;
-	case REPORT_ID_BOOT_KEYBOARD:
-		report_data_id = REPORT_ID_KEYBOARD_KEYS;
-		break;
-	default:
-		/* Use the same report id. */
-		break;
-	}
-
-	struct report_data *rd = get_report_data(report_data_id);
+	struct report_data *rd = rs_get_linked_rd(report_id);
```

Comment:
`rs_get_linked_rd` could return `NULL`. You should check that before referencing the pointer in line `1113`.

Reply:
This should not happen. The old code was essentially the same. Now code suggests we can exit with `rs` structure partly filled in :(
If we care to add a check we should use assert instead of WRN and return.

Reply:
this also need to be changed to assert

Reply:
> assert INSTEAD of WRN and return

Assert is NOT an error handler. It is added when you are sure something cannot occur. Error handlers should be added when you expect that something can happen and need to be properly services.
The code under the condition is not a proper error handler. I don't think expanding it to one is justified (as the situation cannot happen) so it's better to simply remove it.

BTW
Some people are against asserting for NULL pointers. This is well justified as NULL pointer dereference will lead to segfault. So there is no point adding an assert as you will have a free check already. Why check is free - because you don't need to maintain it.
This is why there was no assert here - you have dereference at line 1134. 
But some people like to be explicit and would gladly add such things whenever possible.